src/libtrx/game/music/common.c

```diff
@@ -205,13 +231,15 @@ bool Music_Play(const MUSIC_TRACK_ID track_id, const MUSIC_PLAY_MODE mode)
     }
 
     M_SyncVolume(m_AudioStreamID);
-    Audio_Stream_SetIsLooped(m_AudioStreamID, mode == MPM_LOOPED);
+    Audio_Stream_SetIsLooped(m_AudioStreamID, is_looped);
     Audio_Stream_SetFinishCallback(m_AudioStreamID, M_StreamFinished, nullptr);
 
 finish:
     m_TrackDelayed = MX_INACTIVE;
-    if (mode == MPM_LOOPED) {
+    if (is_looped) {
+        m_TrackCurrent = MX_INACTIVE;
```

Comment:
Wait, why do we set this to `MX_INACTIVE` if we actually play a track `track_id`?

Reply:
This is needed because `Music_GetCurrentPlayingTrack` determines if the current track is `MX_INACTIVE` then it should return the looped track. Otherwise you could cross a normal trigger, then an ambient trigger, save and load, and the regular music would load in rather than the ambient.

Reply:
Thanks, the latter comment is pivotal. Let's include this explanation in the code :relieved:

Reply:
Added 😌